lib/dynasm-backend/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+#![feature(proc_macro_hygiene)]
+
+#[cfg(not(any(
+    all(target_os = "macos", target_arch = "x86_64"),
+    all(target_os = "linux", target_arch = "x86_64"),
+)))]
+compile_error!("This crate doesn't yet support compiling on operating systems other than linux and macos and architectures other than x86_64");
+
+extern crate dynasmrt;
+
+#[macro_use]
+extern crate dynasm;
+
+#[macro_use]
+extern crate lazy_static;
+
+extern crate byteorder;
+
+mod codegen;
+mod codegen_x64;
+mod parse;
+mod protect_unix;
+mod stack;
+
+use crate::codegen::{CodegenError, ModuleCodeGenerator};
+use crate::parse::LoadError;
+use std::ptr::NonNull;
+use wasmer_runtime_core::{
+    backend::{sys::Memory, Backend, CacheGen, Compiler, FuncResolver, Token},
+    cache::{Artifact, Error as CacheError},
+    error::{CompileError, CompileResult},
+    module::{ModuleInfo, ModuleInner},
+    types::LocalFuncIndex,
+    vm,
+};
+
+struct Placeholder;
+impl CacheGen for Placeholder {
+    fn generate_cache(
+        &self,
+        _module: &ModuleInner,
+    ) -> Result<(Box<ModuleInfo>, Box<[u8]>, Memory), CacheError> {
+        Err(CacheError::Unknown(
+            "the dynasm backend doesn't support caching yet".to_string(),
+        ))
+    }
+}
+
+impl FuncResolver for Placeholder {
+    fn get(
+        &self,
+        _module: &ModuleInner,
+        _local_func_index: LocalFuncIndex,
+    ) -> Option<NonNull<vm::Func>> {
+        NonNull::new(0x3f3f3f3f3f3f3f3fusize as *mut vm::Func)
+    }
+}
+
+pub struct SinglePassCompiler {}
+impl SinglePassCompiler {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Compiler for SinglePassCompiler {
+    fn compile(&self, wasm: &[u8], _: Token) -> CompileResult<ModuleInner> {
+        let mut mcg = codegen_x64::X64ModuleCodeGenerator::new();
```

Comment:
Can we cfg this to error at compile-time if the target arch isn't supported?

Reply:
It is already checked at L3-L7.

Reply:
Okay.